src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GenericLookupResult.cs

```diff
@@ -103,4 +103,60 @@ public override string GetMangledName(NameMangler nameMangler)
 
         public override string ToString() => $"VirtualCall: {_method}";
     }
+
+    /// <summary>
+    /// Generic lookup result that points to the non-GC static base of a type.
+    /// </summary>
+    internal sealed class TypeNonGCStaticBaseGenericLookupResult : GenericLookupResult
+    {
+        private MetadataType _type;
+
+        public TypeNonGCStaticBaseGenericLookupResult(TypeDesc type)
+        {
+            Debug.Assert(type.IsRuntimeDeterminedSubtype, "Concrete static base in a generic dictionary?");
+            Debug.Assert(type is MetadataType);
+            _type = (MetadataType)type;
+        }
+
+        public override ISymbolNode GetTarget(NodeFactory factory, Instantiation typeInstantiation, Instantiation methodInstantiation)
+        {
+            var instantiatedType = (MetadataType)_type.InstantiateSignature(typeInstantiation, methodInstantiation);
+            return factory.TypeNonGCStaticsSymbol(instantiatedType);
```

Comment:
Note to self: this is wrong because `TypeNonGCStaticsSymbol` will point to the class constructor context address if there's a lazy cctor, no matter what the name says. This wasn't always the case (introduced in #1022). We really should just fix that...